lib/components/form/date-time-modal.tsx

```diff
@@ -0,0 +1,113 @@
+// TODO: TypeScript with props.
+/* eslint-disable react/prop-types */
+import { connect } from 'react-redux'
+import coreUtils from '@opentripplanner/core-utils'
+import React, { useCallback } from 'react'
+
+import { AppConfig } from '../../util/config-types'
+import { AppReduxState, FilterType, SortType } from '../../util/state-types'
+import { setQueryParam } from '../../actions/form'
+
+import { StyledDateTimeSelector } from './styled'
+import { updateItineraryFilter } from '../../actions/narrative'
+
+type Props = {
+  config: AppConfig
+  date: string
+  dateFormatLegacy?: string
+  departArrive: DepartArriveValue
+  setQueryParam: (params: any) => void
+  sort: SortType
+  time: string
+  timeFormatLegacy?: string
+  updateItineraryFilter: (payload: FilterType) => void
+}
+
+type DepartArriveValue = 'NOW' | 'DEPART' | 'ARRIVE'
+
+function DateTimeModal(props: Props) {
+  const {
+    config,
+    date,
+    dateFormatLegacy,
+    departArrive,
+    setQueryParam,
+    sort,
+    time,
+    timeFormatLegacy,
+    updateItineraryFilter
+  } = props
+  const { homeTimezone, isTouchScreenOnDesktop } = config
+  const touchClassName = isTouchScreenOnDesktop
+    ? 'with-desktop-touchscreen'
+    : ''
+
+  const syncSortWithDepartArrive = config?.itinerary?.syncSortWithDepartArrive
+  const setQueryParamMiddleware = useCallback(
+    (params: any) => {
+      if (syncSortWithDepartArrive !== false) {
+        switch (params.departArrive) {
+          case 'NOW':
+            updateItineraryFilter({ sort: { ...sort, type: 'DURATION' } })
+            break
+          case 'DEPART':
+            updateItineraryFilter({ sort: { ...sort, type: 'DEPARTURETIME' } })
+            break
+          case 'ARRIVE':
+            updateItineraryFilter({ sort: { ...sort, type: 'ARRIVALTIME' } })
+            break
```

Comment:
In my dreams I'm imagining a pretty map that we can use to avoid some of this repeat code, but since I don't think this will change much I think it's fine

Reply:
good idea!